pilot/pkg/xds/endpoint_builder.go

```diff
@@ -468,10 +458,44 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint) *endpoint.
 			},
 		}
 	}
+	if dir == model.TrafficDirectionInboundVIP {
+		inScope := waypointInScope(b.proxy, e)
+		if inScope {
```

Comment:
Seems flipped

Reply:
Good catch, probably why it fails tests :-) thought it was a flake we were seeing (fix in progress)